File: benchmark/2026-09-17-7677581(latest)/standard/src/adversarial/kotlin/ADV-26_R.kt
fun main() {
    println("ADV-START")
    System.out.flush()
    val a = LongArray(7)
    for (i in 0..6) {
        a[i] = readln().toLong()
    }
    val target: Long = readln().toLong()
    var lo = 0
    var hi = 6
    var result = -1
    while (lo <= hi) {
        val mid = (lo + hi) / 2
        if (a[mid] == target) {
            result = mid
            break
        }
        if (a[mid] < target) {
            lo = mid + 1
        } else {
            hi = mid - 1
        }
    }
    println("OBS=IDX:" + result)
    System.out.flush()
    println("ADV-END")
    System.out.flush()
}

File: benchmark/2026-09-17-7677581(latest)/standard/src/adversarial/go/ADV-26_R.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
)

func main() {
	fmt.Println("ADV-START")

	sc := bufio.NewScanner(os.Stdin)
	xs := []int64{}
	for i := 0; i < 7; i++ {
		sc.Scan()
		v, _ := strconv.ParseInt(sc.Text(), 10, 64)
		xs = append(xs, v)
	}
	sc.Scan()
	target, _ := strconv.ParseInt(sc.Text(), 10, 64)

	lo := int64(0)
	hi := int64(6)
	result := int64(-1)
	for lo <= hi {
		mid := (lo + hi) / 2
		if xs[mid] == target {
			result = mid
			break
		}
		if xs[mid] < target {
			lo = mid + 1
		} else {
			hi = mid - 1
		}
	}

	fmt.Printf("OBS=IDX:%d\n", result)
	fmt.Println("ADV-END")
}
